scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5240 @@
+    True
+
+    Pseudo-random samples can be drawn from
+    the underlying distribution using ``sample``.
+
+    >>> X.rng = np.random.default_rng(2354873452)
+    >>> X.sample(shape=(4,))
+    {repr(X.sample(shape=(4,)))}
+    """
+    # remove the indentation due to use of block quote within function;
+    # eliminate blank first line
+    example = "\n".join([line.lstrip() for line in example.split("\n")][1:])
+    return example
+
+
+class ContinuousDistribution:
+    r""" Class that represents a continuous statistical distribution.
+
+    Instances of the class represent a random variable.
+
+    Parameters
+    ----------
+    tol : positive float, optional
+        The desired relative tolerance of calculations. Left unspecified,
+        calculations may be faster; when provided, calculations may be
+        more likely to meet the desired accuracy.
+    validation_policy : {None, "skip_all"}
+        Specifies the level of input validation to perform. Left unspecified,
+        input validation is performed to ensure appropriate behavior in edge
+        case (e.g. parameters out of domain, argument outside of distribution
+        support, etc.) and improve consistency of output dtype, shape, etc.
+        Pass ``'skip_all'`` to avoid the computational overhead of these
+        checks when rough edges are acceptable.
+    cache_policy : {None, "no_cache"}
+        Specifies the extent to which intermediate results are cached. Left
+        unspecified, intermediate results of some calculations (e.g. distribution
+        support, moments, etc.) are cached to improve performance of future
+        calculations. Pass ``'no_cache'`` to reduce memory reserved by the class
+        instance.
+    rng : numpy.random.Generator
+        Random number generator to be used by any methods that require
+        pseudo-random numbers (e.g. `sample`).
+
+    Attributes
+    ----------
+    All parameters are available as attributes.
+
+    Methods
+    -------
+    support
+
+    plot
+
+    sample
+
+    moment
+
+    mean
+    median
+    mode
+
+    variance
+    standard_deviation
+
+    skewness
+    kurtosis
+
+    pdf
+    logpdf
+
+    cdf
+    icdf
+    ccdf
+    iccdf
+
+    logcdf
+    ilogcdf
+    logccdf
+    ilogccdf
+
+    entropy
+    logentropy
+
+    Notes
+    -----
+    The following abbreviations are used throughout the documentation.
+
+    - PDF: probability density function
+    - CDF: cumulative distribution function
+    - CCDF: complementary CDF
+    - entropy: differential entropy
+    - log-*F*: logarithm of *F* (e.g. log-CDF)
+    - inverse *F*: inverse function of *F* (e.g. inverse CDF)
+
+    The API documentation is written to describe the API, not to serve as
+    a statistical reference. Effort is made to be correct at the level
+    required to use the functionality, not to be mathematically rigorous.
+    For example, continuity and differentiability may be implicitly assumed.
```

Comment:
Hm I probably didn't explain it very well, but my point wasn't about the documentation. 

I was trying to suggest that perhaps we could use e.g. the `point` param of `scipy.integrate.quad` to let the routine know where the discontinuities and such are, so it can avoid it. 

So that's what I had in mind when I was talking about "adding a property that returns a tuple of the discontinuities on the domain", which in hindsight, wasn't as clear as I initially hoped it would be 😅

Reply:
> Hm I probably didn't explain it very well, but my point wasn't about the documentation.

I see now. I was confused mostly because this referenced line is a sentence in the documentation that refers to_the documentation, not the implementation. For implementation, we do not need to make those assumptions. Such a feature can certainly be added when it is needed.